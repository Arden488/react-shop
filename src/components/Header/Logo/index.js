import React, { Component } from 'react';
import feather from 'feather-icons';

class Logo extends Component {
  componentDidMount() {
    feather.replace();
  }

  render() {
    return (
      <div>
        <i data-feather="zap" />
      </div>
    );
  }
}

export default Logo;
