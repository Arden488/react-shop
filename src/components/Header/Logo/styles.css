.header-logo a {
  color: #fff;
}

.header-logo svg {
  display: inline-block;
  vertical-align: middle;
}

.header-logo span {
  display: inline-block;
  vertical-align: middle;
  margin-left: 10px;
}