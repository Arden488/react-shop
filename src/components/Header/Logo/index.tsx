import * as feather from 'feather-icons';
import * as React from 'react';
import { Link } from 'react-router-dom';

import './styles.css';

class Logo extends React.Component {
  public componentDidMount() {
    feather.replace();
  }

  public render() {
    return (
      <div className="header-logo">
        <Link to="/">
          <i data-feather="zap" />
          <span>ZapShop</span>
        </Link>
      </div>
    );
  }
}

export default Logo;
