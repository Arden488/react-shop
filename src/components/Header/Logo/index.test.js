/* global it, expect, beforeEach */

import React from 'react';
import { shallow } from 'enzyme';
import Logo from './index';

let component = null;

beforeEach(() => {
  component = shallow(<Logo />);
});

it('renders without crashing', () => {
  expect(component.exists()).toEqual(true);
});
