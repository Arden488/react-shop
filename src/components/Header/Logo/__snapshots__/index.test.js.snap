// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="header-logo"
>
  <Link
    replace={false}
    to="/"
  >
    <i
      data-feather="zap"
    />
    <span>
      ZapShop
    </span>
  </Link>
</div>
`;
